### PR TITLE
[INTEG-766] SAP Frontend connects to backend with AIR header

### PR DESCRIPTION
**Pull request by ryunsong-contentful**
## Purpose
For demo purposes. Lots of code changes and updates to get this into a workable states.

Everything done in this PR enables the frontend to fetch products and list them from your SAP Store using the `Application-interface-key fyi` header.

The frontend needs to be re-written and updated.

## Approach
Do whatever it takes to make this app demoable

## Testing steps
Test in the UI and app

## Breaking Changes
None 

## Dependencies and/or References
None

## Deployment
Standard

### CODE REVIEW

File: apps/sap-commerce-cloud-with-air/frontend/package.json [hunk 1]
```diff
@@ -44,6 +45,7 @@
     ]
   },
   "devDependencies": {
+    "@babel/plugin-proposal-private-property-in-object": "^7.21.11",
```
Comment (whitelisab) [inline apps/sap-commerce-cloud-with-air/frontend/package.json:48]: What is this being used for?

File: apps/sap-commerce-cloud-with-air/frontend/src/AppConfig/AppConfig.tsx [hunk 2]
```diff
@@ -192,35 +192,37 @@ export default class AppConfig extends React.Component<Props, State> {
...
-                      width: def.type === 'Symbol' ? 'large' : 'medium',
-                      type: def.type === 'Symbol' ? 'text' : 'number',
-                      maxLength: 255,
-                    }}
-                    helpText={def.description}
-                    value={parameters[def.id]}
-                    onChange={this.onParameterChange.bind(this, def.id)}
-                  />
+                  <div style={{ marginBottom: '15px' }}>
+                    <TextField
```
Comment (whitelisab) [inline apps/sap-commerce-cloud-with-air/frontend/src/AppConfig/AppConfig.tsx:202]: Some of the frontend issues are probably due to the fact that these are deprecated components from Forma36, we can fix in a follow up PR.

File: apps/sap-commerce-cloud-with-air/frontend/src/components/DialogClass.tsx [hunk 3]
```diff
@@ -281,13 +288,15 @@ export default class Dialog extends React.Component<DialogProps, State> {
...
         </div>
         {isFieldTypeArray ? (
-          <Button
-            buttonType="primary"
-            icon="Done"
-            onClick={this.selectMultipleProductsClickEvent}
-            style={{ margin: '20px' }}>
-            Select Products
-          </Button>
+          <div>
```
Comment (whitelisab) [inline apps/sap-commerce-cloud-with-air/frontend/src/components/DialogClass.tsx:291]: Are you just commenting out this button temporarily? Maybe add a TODO to enable it again?

**Review by whitelisab — COMMENTED** [review]
Looks like a good start! I added a few questions for you on this PR, but going to approve it so you can keep moving on this work.

**Review by whitelisab — APPROVED** [review]

File: apps/sap-commerce-cloud-with-air/frontend/src/AppConfig/AppConfig.tsx (hunk 2 above)
Comment (ryunsong-contentful) [reply apps/sap-commerce-cloud-with-air/frontend/src/AppConfig/AppConfig.tsx:202]: Yup that's what i noticed as well

File: apps/sap-commerce-cloud-with-air/frontend/src/api/fetchBaseSites.ts [hunk 4]
```diff
@@ -1,28 +1,20 @@
...
 import { config } from '../config';
 import { SAPParameters } from '../interfaces';
+// import { baseSitesMock } from './realMockData';
+import fetchWithSignedRequest from './signed-requests';
+import { DialogAppSDK } from '@contentful/app-sdk';
 
 export async function fetchBaseSites(
   parameters: SAPParameters,
-  applicationInterfaceKey: string
+  applicationInterfaceKey: string,
```
[the remark this answers is not in the document]
Comment (ryunsong-contentful) [reply apps/sap-commerce-cloud-with-air/frontend/src/api/fetchBaseSites.ts:9]: Yup, this PR is a bit of a mess

File: apps/sap-commerce-cloud-with-air/frontend/src/components/DialogClass.tsx (hunk 3 above)
Comment (ryunsong-contentful) [reply apps/sap-commerce-cloud-with-air/frontend/src/components/DialogClass.tsx:291]: Yes i will do that now in a separate follow up branch

**Pull request merged**